pkg/client/client.go

```diff
@@ -70,28 +59,107 @@ func New(config *ConmonServerConfig) (_ *ConmonClient, retErr error) {
 			cl.Shutdown()
 		}
 	}()
-	if err := os.Remove(config.ConmonPIDFile); err != nil {
+	if err := cl.waitUntilServerUp(); err != nil {
+		return nil, err
+	}
+	if err := os.Remove(cl.pidFile()); err != nil {
 		return nil, err
 	}
 	return cl, nil
 }
 
+func (c *ConmonServerConfig) ToClient() (*ConmonClient, error) {
+	if err := os.MkdirAll(c.ServerRunDir, 0o755); err != nil && !os.IsExist(err) {
+		return nil, fmt.Errorf("couldn't create run dir %s", c.ServerRunDir)
+	}
+
+	return &ConmonClient{
+		runDir: c.ServerRunDir,
+	}, nil
+}
+
+func (c *ConmonClient) StartServer(config *ConmonServerConfig) error {
+	entrypoint, args, err := c.toArgs(config)
+	if err != nil {
+		return err
+	}
+	cmd := exec.Command(entrypoint, args...)
+	if config.Stdin != nil {
```

Comment:
This config stdio block was in the original code, but it appears like it should be assigning config.Stdin and friends... perhaps defaulting the os.Std* first

Reply:
oh yeah this is definitely messed up haha

Reply:
fixed